parallel/runner.go

```diff
@@ -57,6 +60,14 @@ type runner struct {
 	activeThreads uint32
 	// The number of tasks in the queue.
 	totalTasksInQueue uint32
+	// Indicate that the runner has finished.
+	finishedNotifier chan bool
+	// Indicates that the finish channel is closed.
+	finishedNotifierChannelClosed bool
+	// A lock for the finishedNotifier check.v
+	finishedNotifierLock sync.Mutex
+	// A flag that enables the finishedNotifier operation.
```

Comment:
A flag that enables the finishedNotifier operation.
-->
A flag that allows receiving a notification through a channel, when the runner finishes executing all the tasks.